src/main/java/org/opensearch/performanceanalyzer/PerformanceAnalyzerPlugin.java

```diff
@@ -189,27 +195,61 @@ public PerformanceAnalyzerPlugin(final Settings settings, final java.nio.file.Pa
         clusterSettingsManager.addSubscriberForIntSetting(
                 PerformanceAnalyzerClusterSettings.PA_NODE_STATS_SETTING, nodeStatsSettingHandler);
 
+        // Adding RTF Collectors if flag is enabled in performance-analyzer.properties
+        if (PluginSettings.instance().isTelemetryCollectorsEnabled()) {
+            LOG.info("Scheduling Telemetry Collectors");
+            scheduleTelemetryCollectors();
+        }
+
+        // Adding RCA Collectors if flag is enabled in performance-analyzer.properties
+        if (PluginSettings.instance().isRcaCollectorsEnabled()) {
+            LOG.info("Scheduling RCA Collectors");
+            scheduleRcaCollectors();
+        }
+
+        scheduledMetricCollectorsExecutor.start();
+
+        EventLog eventLog = new EventLog();
```

Comment:
Can we do this also behind a flag?

Reply:
We should not do it since we are now dynamically controlling collectors. EventLog should be always initialised if we wish to control the changes dynamically, otherwise it will require a B/G.